src/script/history.ts

```diff
@@ -0,0 +1,130 @@
+// Constants for localStorage keys
+const HISTORY_KEY = 'calculationHistory';
+
+// Interface for history item type
+interface HistoryItem {
+    operation: string;
+    result: string;
+}
+
+// save-history
+export const saveHistory = (historyEntry: string): void => {
+    let history: HistoryItem[] = [];
+    try {
+        history = JSON.parse(localStorage.getItem(HISTORY_KEY) || '[]');
+    } catch (error) {
+        console.error("Error parsing history from localStorage", error);
+    }
+
+    // Extract operation and result
+    const [operation, result] = historyEntry.split('=');
+    
+    // Save both operation and result
+    const historyItem: HistoryItem = { operation: operation.trim(), result: result.trim() };
+    history.push(historyItem);
+    localStorage.setItem(HISTORY_KEY, JSON.stringify(history));
+};
+
+// clear-history
+export const clearHistory = (): void => {
+    localStorage.removeItem(HISTORY_KEY);
+    const historyList = document.getElementById('historyList') as HTMLElement;
+    const historyContainer = document.getElementById('historyContainer') as HTMLElement;
+
+    if (historyList) {
+        historyList.innerHTML = '';
+    }
+
+    if (historyContainer) {
+        // Show "No history available" message after clearing history
+        const noHistoryMessage = document.createElement('p');
+        noHistoryMessage.textContent = 'No history available';
+        historyList.appendChild(noHistoryMessage);
+    }
+};
+
+// display-history
+export const displayHistory = (): void => {
+    let history: HistoryItem[] = [];
+    try {
+        history = JSON.parse(localStorage.getItem(HISTORY_KEY) || '[]');
+    } catch (error) {
+        console.error("Error parsing history from localStorage", error);
+    }
+
+    const historyList = document.getElementById('historyList') as HTMLElement;
+    const historyContainer = document.getElementById('historyContainer') as HTMLElement;
+    const toggleHistoryButton = document.getElementById('toggleHistoryButton') as HTMLElement;
+
+    if (!historyList || !historyContainer) {
+        console.error("History list or container element not found.");
+        return;
+    }
+
+    // Display history content
+    renderHistoryList(history, historyList);
+    toggleHistoryDisplay(historyContainer, toggleHistoryButton);
+};
+
+// Render history list
+const renderHistoryList = (history: HistoryItem[], historyList: HTMLElement): void => {
```

Comment:
Nit: Can change the other similar instances and update it.

```suggestion
const renderHistoryList = (history: Array<HistoryItem>, historyList: HTMLElement): void => {
```

Reply:
solved https://github.com/Sakshi-Simform/calculator-ts/pull/2/commits/c71c8be8084cb33ad54ec31e9cabe448945d11f1

Reply:
Have we checked the all the instances as mentioned in the above comments?

Reply:
@Sakshi-Simform ^^^

Reply:
fixed https://github.com/Sakshi-Simform/calculator-ts/pull/2/commits/7b7ed0f3f849a181068491d28d1e644fe2b12cb6